.github/workflows/test-and-deploy.yml

```diff
@@ -0,0 +1,59 @@
+name: Test and Deploy
+on: push
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: ruby/setup-ruby@v1
+      - name: Install gems
+        run: |
+          bundle install
+      - run:
+          bundle exec rspec
+
+  deploy-lambda-qa:
+    runs-on: ubuntu-latest
+    permissions: 
+      id-token: write
+      contents: read
+    needs: test
+    if: github.ref == 'refs/head/qa'
```

Comment:
I've been using `tgr-qa` as a temporary measure but this is fine if you are comfortable using the regular `qa` branch

Reply:
I think this should be ok to merge directly and then test or do you prefer testing the PR and then merging it? We'd have to update the lambda to use the ruby 3.3 runtime, but if it fails, I don't think we can revert to a lower ruby runtime.